internal/cli/atlas/clusters/watch.go

```diff
@@ -47,6 +49,14 @@ func (opts *WatchOpts) watcher() (bool, error) {
 	if err != nil {
 		return false, err
 	}
+	if result.StateName == "UPDATING" {
+		trigger := func(err error) bool {
+			var atlasErr *atlas.ErrorResponse
+			errorCode := "CLUSTER_NOT_FOUND"
+			return errors.As(err, &atlasErr) && atlasErr.ErrorCode == errorCode
+		}
```

Comment:
[nit] every time you loop you'll be redefining the function and you don't need closures 

```go
func isRetryable(err error) bool {
	var atlasErr *atlas.ErrorResponse
	return errors.As(err, &atlasErr) && atlasErr.ErrorCode == "CLUSTER_NOT_FOUND"
}

func (opts *WatchOpts) watcher() (bool, error) {
	result, err := opts.store.AtlasCluster(opts.ConfigProjectID(), opts.name)
	if err != nil {
		return false, err
	}
	if result.StateName == "UPDATING" {
		opts.IsRetryableErr = isRetryable
	}
	return result.StateName == "IDLE", nil
}
```